test/e2e/suite/nodeadm/nodeadm_test.go

```diff
@@ -0,0 +1,183 @@
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v2"
+
+	"github.com/aws/eks-hybrid/test/e2e"
+	"github.com/aws/eks-hybrid/test/e2e/credentials"
+	"github.com/aws/eks-hybrid/test/e2e/kubernetes"
+	"github.com/aws/eks-hybrid/test/e2e/nodeadm"
+	osystem "github.com/aws/eks-hybrid/test/e2e/os"
+	"github.com/aws/eks-hybrid/test/e2e/suite"
+)
+
+var (
+	filePath    string
+	suiteConfig *suite.SuiteConfiguration
+)
+
+func init() {
+	flag.StringVar(&filePath, "filepath", "", "Path to configuration")
+}
+
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Nodeadm E2E Suite")
+}
+
+var _ = SynchronizedBeforeSuite(
+	// This function only runs once, on the first process
+	// Here is where we want to run the setup infra code that should only run once
+	// Whatever information we want to pass from this first process to all the processes
+	// needs to be serialized into a byte slice
+	// In this case, we use a struct marshalled in json.
+	func(ctx context.Context) []byte {
+		suiteConfig := suite.BeforeSuiteCredentialSetup(ctx, filePath)
+		suiteJson, err := yaml.Marshal(suiteConfig)
+		Expect(err).NotTo(HaveOccurred(), "suite config should be marshalled successfully")
+		return suiteJson
+	},
+	// This function runs on all processes, and it receives the data from
+	// the first process (a json serialized struct)
+	// The only thing that we want to do here is unmarshal the data into
+	// a struct that we can make accessible from the tests. We leave the rest
+	// for the per tests setup code.
+	func(ctx context.Context, data []byte) {
+		suiteConfig = suite.BeforeSuiteCredentialUnmarshal(ctx, data)
+	},
+)
+
+var _ = Describe("Hybrid Nodes", func() {
+	notSupported := suite.NodeadmConfigMatchers{
+		{
+			MatchOS:            osystem.IsUbuntu2004,
+			MatchCredsProvider: credentials.IsIAMRolesAnywhere,
+		},
+		{
+			MatchOS:            osystem.IsRHEL8,
+			MatchCredsProvider: credentials.IsIAMRolesAnywhere,
+		},
+	}
+
+	When("using peered VPC", func() {
+		var test *suite.PeeredVPCTest
+		osList := suite.OSList()
+		credentialProviders := suite.CredentialProviders()
+
+		// Here is where we setup everything we need for the test. This includes
+		// reading the setup output shared by the "before suite" code. This is the only place
+		// that should be reading that global state, anything needed in the test code should
+		// be passed down through "local" variable state. The global state should never be modified.
+		BeforeEach(func(ctx context.Context) {
+			test = suite.BeforeVPCTest(ctx, suiteConfig)
+			credentialProviders = suite.AddClientsToCredentialProviders(credentialProviders, test)
+		})
+
+		When("using ec2 instance as hybrid nodes", func() {
+			upgradeEntries := []TableEntry{}
+			initEntries := []TableEntry{}
+			for _, nodeOS := range osList {
+			providerLoop:
+				for _, provider := range credentialProviders {
+					if notSupported.Matches(nodeOS.Name(), provider.Name()) {
+						continue providerLoop
+					}
```

Comment:
don't we need to do this in the conformance tests as well?

Something we could do is have a shared method that generates all valid combinations of Os and provider and returns the list of tuples. Then from each suite you decide if you use all of them or a subset.